docs/architecture/maui/accessing-remote-data.md

```diff
@@ -1,9 +1,380 @@
+
+For more information about REST, see [API design](/azure/architecture/best-practices/api-design) and [API implementation](/azure/architecture/best-practices/api-implementation) on Microsoft Docs.
+
+## Consuming RESTful APIs
+
+The eShopOnContainers mobile app uses the Model-View-ViewModel (MVVM) pattern, and the model elements of the pattern represent the domain entities used in the app. The controller and repository classes in the eShopOnContainers reference application accept and return many of these model objects. Therefore, they are used as data transfer objects (DTOs) that hold all the data that is passed between the mobile app and the containerized microservices. The main benefit of using DTOs to pass data to and receive data from a web service is that by transmitting more data in a single remote call, the app can reduce the number of remote calls that need to be made.
+
+## Making web requests
+
+The eShopOnContainers mobile app uses the `HttpClient` class to make requests over HTTP, with JSON being used as the media type. This class provides functionality for asynchronously sending HTTP requests and receiving HTTP responses from a URI identified resource. The HttpResponseMessage class represents an HTTP response message received from a REST API after an HTTP request has been made. It contains information about the response, including the status code, headers, and any body. The HttpContent class represents the HTTP body and content headers, such as Content-Type and Content-Encoding. The content can be read using any of the `ReadAs` methods, such as `ReadAsStringAsync` and `ReadAsByteArrayAsync`, depending on the format of the data.
+
+## Making a GET request
+
+The `CatalogService` class is used to manage the data retrieval process from the catalog microservice. In the `RegisterViewModels` method in the `MauiProgram` class, the `CatalogService` class is registered as a type mapping against the `ICatalogService` type with the dependency injection container. Then, when an instance of the `CatalogViewModel` class is created, its constructor accepts an `ICatalogService type`, which the dependency injection container resolves, returning an instance of the `CatalogService` class. For more information about dependency injection, see [Dependency Injection](dependency-injection.md).
+
+The image below shows the interaction of classes that read catalog data from the catalog microservice for displaying by the CatalogView.
+
+:::image type="content" source="media/retrieving-data-for-catalog.png" alt-text="Retrieving data from the catalog microservice.":::
+
+When the `CatalogView` is navigated to, the `OnInitialize` method in the CatalogViewModel class is called. This method retrieves catalog data from the catalog microservice, as demonstrated in the following code example:
+
+```csharp
+public override async Task InitializeAsync(object navigationData)
```

Comment:
If the `navigationData` is not going to be used, consider either discarding it with the `_`, or add a code comment to explicitly state that it's not needed, or perhaps introduced later.